src/gui/plugins/transform_control_logic/TransformControlLogic.cc

```diff
@@ -0,0 +1,738 @@
+
+  /// \brief Whether the transform gizmo is being dragged.
+  public: bool transformActive{false};
+
+  /// \brief Transform mode service
+  public: std::string transformModeService;
+
+  /// \brief Transport node
+  public: transport::Node node;
+
+  /// \brief Name of service for setting entity pose
+  public: std::string poseCmdService;
+
+  /// \brief Name of the world
+  public: std::string worldName;
+
+  /// \brief Transform mode: none, translation, rotation, or scale
+  public: rendering::TransformMode transformMode =
+      rendering::TransformMode::TM_NONE;
+
+  /// \brief Transform space: local or world
+  public: rendering::TransformSpace transformSpace =
+      rendering::TransformSpace::TS_LOCAL;
+
+  /// \brief Currently selected entities, organized by order of selection.
+  public: std::vector<Entity> selectedEntities;
+
+  /// \brief Transform controller for models
+  public: rendering::TransformController transformControl;
+
+  public: ignition::common::MouseEvent mouseEvent;
+
+  public: bool mouseDirty = false;
+
+  /// \brief Flag to indicate whether the x key is currently being pressed
+  public: bool xPressed = false;
+
+  /// \brief Flag to indicate whether the y key is currently being pressed
+  public: bool yPressed = false;
+
+  /// \brief Flag to indicate whether the z key is currently being pressed
+  public: bool zPressed = false;
+
+  /// \brief Where the mouse left off - used to continue translating
+  /// smoothly when switching axes through keybinding and clicking
+  /// Updated on an x, y, or z, press or release and a mouse press
+  public: math::Vector2i mousePressPos = math::Vector2i::Zero;
+
+  /// \brief Flag to keep track of world pose setting used
+  /// for button translating.
+  public: bool isStartWorldPosSet = false;
+
+  /// \brief The starting world pose of a clicked visual.
+  public: ignition::math::Vector3d startWorldPos = math::Vector3d::Zero;
+
+  public: ignition::common::KeyEvent keyEvent;
+
+  public: std::mutex mutex;
+
+  //// \brief Pointer to the rendering scene
+  public: rendering::ScenePtr scene = nullptr;
+
+  /// \brief User camera
+  public: rendering::CameraPtr camera = nullptr;
+
+  /// \brief The xyz values by which to snap the object.
+  public: math::Vector3d xyzSnap = math::Vector3d::One;
+
+  /// \brief The rpy values by which to snap the object.
+  public: math::Vector3d rpySnap = {45, 45, 45};
+
+  /// \brief The scale values by which to snap the object.
+  public: math::Vector3d scaleSnap = math::Vector3d::One;
+
+  public: bool blockOrbit = false;
+};
+
+using namespace ignition;
+using namespace gazebo;
+using namespace plugins;
+
+void TransformControlLogicPrivate::HandleTransform()
+{
+  if (nullptr == this->scene)
+  {
+    this->scene = rendering::sceneFromFirstRenderEngine();
+    if (nullptr == this->scene)
+      return;
+
+    this->camera = std::dynamic_pointer_cast<rendering::Camera>(
+      this->scene->SensorByName("Scene3DCamera"));
```

Comment:
I commented on another plugin, but I think we should think carefully about hardcoding the camera's name.